edx_user_state_client/tests.py

```diff
@@ -352,6 +353,47 @@ def test_delete_many_last_field(self):
         self.delete_many(user=0, blocks=[0, 1], fields=['a', 'b'])
         self.assertItemsEqual(self.get_many(user=0, blocks=[0, 1]), [])
 
+    def test_get_mod_date(self):
+        start_time = datetime.now(pytz.utc)
+        self.set_many(user=0, block_to_state={0: {'a': 'b'}, 1: {'b': 'c'}})
+        end_time = datetime.now(pytz.utc)
+
+        mod_dates = self.get(user=0, block=0)
+
+        self.assertItemsEqual(mod_dates.state.keys(), ["a"])
+        self.assertGreater(mod_dates.updated, start_time)
+        self.assertLess(mod_dates.updated, end_time)
+
+    def test_get_many_mod_date(self):
+        start_time = datetime.now(pytz.utc)
+        self.set_many(
+            user=0,
+            block_to_state={0: {'a': 'b'}})
+        mid_time = datetime.now(pytz.utc)
+        self.set_many(
+            user=0,
+            block_to_state={1: {'a': 'c'}})
+        end_time = datetime.now(pytz.utc)
+
+        mod_dates = list(self.get_many(
+            user=0,
+            blocks=[0, 1],
+            fields=["a"]))
+
+        self.assertItemsEqual(
+            [result.block_key for result in mod_dates],
+            [self._block(0), self._block(1)])
+        self.assertItemsEqual(
+            mod_dates[0].state.keys(),
+            ["a"])
+        self.assertGreater(mod_dates[0].updated, start_time)
+        self.assertLess(mod_dates[0].updated, mid_time)
+        self.assertItemsEqual(
+            mod_dates[1].state.keys(),
+            ["a"])
+        self.assertGreater(mod_dates[1].updated, mid_time)
+        self.assertLess(mod_dates[1].updated, end_time)
```

Comment:
I was thinking that you'd update the same field on the *same* block twice, checking the mod_date via get_many between the set_many calls, to verify that the updated time was marching forward for that single block field.